sources/network/tests/network-test.dylan

```diff
@@ -0,0 +1,174 @@
+assert-no-errors(make(<internet-address>, address: "255.255.255.255"));
+  assert-instance?(<numeric-address>, numeric-host-address(numaddr));
+  // RFC6761 this is guaranteed not to exist
+  assert-signals(<host-not-found>, make(<internet-address>, name: "address.invalid"));
+end test;
+
+define test ipv4-numeric-address-test ()
+  let raw = as(<machine-word>, #x7f000001);
+  let num1 = make(<ipv4-numeric-address>, value: raw, order: #"host-order");
+  let num2 = make(<ipv4-numeric-address>, value: raw, order: #"network-order");
+  assert-instance?(<numeric-address>, num1);
+  assert-instance?(<numeric-address>, num2);
+  assert-equal(raw, host-order(num1));
+  assert-equal(raw, network-order(num2));
+  assert-equal("127.0.0.1", as(<string>, num1))
+end test;
+
+define test ipv6-address-test (expected-failure?: #t)
+  let address = make(<internet-address>, address: "::1");
+  assert-equal("::1", host-address(address));
+end test;
+
+define function tcp-server-one-shot ()
+  with-socket-thread(server?: #t)
+    with-server-socket (srv :: <server-socket>,
+                        port: 8888,
+                        protocol: "tcp")
+      let conn :: <socket> = accept(srv);
+      block (return)
+        while (#t)
+          let data :: <sequence> = read(conn, 16);
+          if (first(data) == ' ')
+            return();
+          else
+            write(conn, data);
+          end if;
+        end while;
+      cleanup
+        close(conn);
+      exception (xep :: <incomplete-read-error>)
+      end block;
+    end;
+  end;    
+end function;
+
+define test tcp-test ()
+  let server :: <thread> = make (<thread>, function: tcp-server-one-shot);
+  sleep(0.5);
+  let conn :: <socket> = make(<socket>, host: "localhost", port: 8888);
+  block ()
+    for (i from 1 to 10)
+      let sent = make(<string>,
+                      size: 16,
+                      fill: as(<character>, 96+i));
+      write(conn, sent);
+      let received = read(conn, 16);
+      assert-equal(sent, received);
+    end for;
+    write(conn, make(<string>, size: 16));
+  cleanup
+    close(conn);
+    join-thread(server);
+  end block;
+end test;
+
+define function udp-server-one-shot()
+  with-socket-thread(server?: #t)
+    with-server-socket (srv :: <server-socket>,
+                        port: 8889,
+                        protocol: "udp")
+      let conn :: <socket> = accept(srv);
+      block (return)
+        while (#t)
+          let data :: <sequence> = read(conn, 1);
+          if (first(data) == ' ')
+            return ();
+          end if;
+          write(conn, data);
+        end while;
+      cleanup
+        close(conn);
+      end block;
+    end;
+  end;    
+end function;
+  
+define test udp-test ()
+  let server :: <thread> = make (<thread>, function: udp-server-one-shot);
+  sleep(0.5);
+  let conn :: <socket> = make(<socket>,
+                              host: "localhost",
+                              port: 8889,
```

Comment:
I guess these tests can fail due to port in use. IIRC we have some existing tests like that also. Let's go with it for now but it might be useful to come up with a free port finder library.